helios-services/src/main/java/com/spotify/helios/master/MasterService.java

```diff
@@ -234,6 +235,14 @@ public MasterService(final MasterConfig config,
       this.agentReaper = null;
     }
 
+    // Set up old job reaper (removes jobs not deployed anywhere and created more than X days ago)
+    if (config.getJobRetention() > 0) {
+      this.oldJobReaper = new OldJobReaper(model, config.getJobRetention());
+    } else {
+      log.info("Reaping of old jobs disabled");
+      this.oldJobReaper = null;
```

Comment:
`Optional<OldJobReaper>` might be nicer than a nullable reference.

Could then be 

```java
oldJobReaper.ifPresent(reaper -> reaper.startAsync().awaitRunning());
```

below (although maybe this is more verbose?)